projects/igniteui-angular/src/lib/grids/grid/grid.component.ts

```diff
@@ -1026,6 +1049,73 @@ export class IgxGridComponent extends IgxGridBaseDirective implements GridType,
         return new IgxGridRow(this, index, rec);
     }
 
+    /**
+     * @hidden @internal
+     */
+    public allRows(): RowType[] {
+        return this.dataView.map((rec, index) => this.createRow(index));
+    }
+
+    /**
+     * Returns the collection of `IgxGridRow`s for current page.
+     *
+     * @hidden @internal
+     */
+    public dataRows(): RowType[] {
+        return this.allRows().filter(row => row instanceof IgxGridRow);
+    }
+
+    /**
+     * Returns an array of the selected `IgxGridCell`s.
+     *
+     * @example
+     * ```typescript
+     * const selectedCells = this.grid.selectedCells;
+     * ```
+     */
+    public get selectedCells(): CellType[] {
```

Comment:
can't we just move those methods under grid-base directive?

Reply:
it wouldn't return quite the correct results, since currently grid-base cannot diferentiate the various row types. can consider it afterwards for enhacement.